src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisCourseChatSessionResource.java

```diff
@@ -0,0 +1,154 @@
+            UserRepository userRepository, CourseRepository courseRepository, IrisSessionService irisSessionService, IrisSettingsService irisSettingsService,
+            PyrisHealthIndicator pyrisHealthIndicator, IrisRateLimitService irisRateLimitService) {
+        this.irisCourseChatSessionRepository = irisCourseChatSessionRepository;
+        this.authCheckService = authCheckService;
+        this.userRepository = userRepository;
+        this.irisSessionService = irisSessionService;
+        this.irisSettingsService = irisSettingsService;
+        this.pyrisHealthIndicator = pyrisHealthIndicator;
+        this.irisRateLimitService = irisRateLimitService;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * GET course-chat/{courseId}/sessions/current: Retrieve the current iris session for the course.
+     *
+     * @param courseId of the course
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the current iris session for the course or {@code 404 (Not Found)} if no session exists
+     */
+    @PostMapping("{courseId}/sessions/current")
+    @EnforceAtLeastStudent
+    public ResponseEntity<IrisCourseChatSession> getCurrentSessionOrCreateIfNotExists(@PathVariable Long courseId) throws URISyntaxException {
+        var course = courseRepository.findByIdElseThrow(courseId);
+
+        irisSettingsService.isEnabledForElseThrow(IrisSubSettingsType.CHAT, course);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
+
+        var sessionOptional = irisCourseChatSessionRepository.findLatestByCourseIdAndUserIdWithMessages(course.getId(), user.getId());
+        if (sessionOptional.isPresent()) {
+            var session = sessionOptional.get();
+
+            // if session is of today we can continue it; otherwise create a new one
+            if (session.getCreationDate().withZoneSameInstant(ZoneId.systemDefault()).toLocalDate().isEqual(LocalDate.now(ZoneId.systemDefault()))) {
+                irisSessionService.checkHasAccessToIrisSession(session, user);
+                return ResponseEntity.ok(session);
+            }
+        }
+
+        // create a new session with an initial message from Iris
+        return createSessionForCourseInternal(courseId, true);
+    }
+
+    /**
+     * GET course-chat/{courseId}/sessions: Retrieve all Iris Sessions for the course
+     *
+     * @param courseId of the course
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body a list of the iris sessions for the course or {@code 404 (Not Found)} if no session exists
+     */
+    @GetMapping("{courseId}/sessions")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<IrisCourseChatSession>> getAllSessions(@PathVariable Long courseId) {
```

Comment:
Is this actually needed? I could not find any usages of this in the client.

Reply:
Not yet, but in the future we can display a chat history